includes/services/extended/github.php

```diff
@@ -25,9 +25,13 @@ function __construct() {
 
 		$creds = $this->get_credentials();
 
```

Comment:
The following were all set on the wpcom version of this service - does anybody know if these are actually needed?

Reply:
It appears that the WPCOM GitHub service has a different origin to the Keyring service (see r75630-wpcom), so I'm inclined to think that this is all WPCOM-specific.

Reply:
Just tested on a local install of Keyring and it doesn't seem to affect the operation of the service at all, so we could either merge it in, or add some WPCOM comments to this and reapply with updates ?

Reply:
I'd like to get @beaulebens' opinion on this one before we move ahead with it.

Reply:
If you're referring to the `$creds` handling , then it's generally there to allow configuration for services through a variety of methods;

https://github.com/beaulebens/keyring/blob/master/service.php#L226

1. if the class has its own `_get_credentials()` use that, or
1. if there are constants for this service use them, or
1. check the database option (`keyring_credentials`) in case they're set there

We have ID, key, and secret just because different services sometimes need all 3. They're definitely not required for all services. It looks like GitHub apps only really have a key and secret, so app_id is probably not very necessary. Doesn't hurt to set it (similar to all other services) though, I don't think. Hope that helps.